objects/Wasp/Step_0.gml

```diff
@@ -0,0 +1,83 @@
+/// @description Boss Movement
+// You can write your code in this editor
+
+function wasp_movement(_speed, _bounce_angle) {
+/// @description  Move around the room randomly avoiding walls and objects
+/// @param  Speed
+/// @param  BounceAngle
+
+	// Define local variables for this script
+	var i;
+	var spd		=	_speed; // The speed to move the object
+	var ang		=	_bounce_angle; // Bounce angle, will prevent us from moving along the walls (I recomend 5)
+
+	var top		=	0;
+	var bottom	=	room_height / 4;
+	var left	=	0; 
+	var right	=	room_width;
+	var hit		=	array_create(5,false); // Here we create an array for our five different types of collisions or "hits"
+
+	// Make sure we are checking colisisons based on center origin
+	var x_middle	= (sprite_width/2);
+	var y_middle	= (sprite_height/2);
+	var prev_offx	= sprite_xoffset;
+	var prev_offy	= sprite_yoffset;
+
```

Comment:
This math can be avoided entirely by setting the origin of the sprite to middle center in s_Wasp

Reply:
I attempted this, but no luck, lol.  The sprite kept disappearing for some reason.  IT works as is though.